cmd/vcluster/cmd/start.go

```diff
@@ -89,7 +89,12 @@ func ExecuteStart(ctx context.Context, options *StartOptions) error {
 	// set features for plugins to recognize
 	plugin.DefaultManager.SetProFeatures(pro.LicenseFeatures())
 
-	// check if we should create certs
+	// connect to vCluster platform if configured
+	startPlatformServersAndControllers, err := pro.ConnectToPlatform(ctx, vConfig)
```

Comment:
Whats the benefit of returning a function here compared to having 2 separate pro functions? I think returning a function always makes the code a bit more complex and I don't really see why this is useful in this case, is this about the closure that is important?

Reply:
@FabianKramm I was not so sure about this myself. The reason why I chose to do it is because the start controller logic is highly dependent on dependencies created above it. Decoupling them would mean returning a handful of variables that I do not think makes sense from a reader's perspective, nor are they useful to other functions. It would also mean introducing a new pro function that I do not think is necessary. There are other solutions but I'm not convinced they are nicer than this. The server/controllers are related to `ConnectToPlatform` at least. 

Another solution that I did consider was starting it in a go routine within `ConnectToPlatform` but blocking it with an empty channel, waitgroup, or some other sync mechanism, and then returning that. Do you think that would be better?